mono/metadata/object.c

```diff
@@ -7831,10 +7831,9 @@ mono_get_eh_callbacks (void)
 void
 mono_raise_exception (MonoException *ex) 
 {
-	/* raise_exception doesn't return, so the transition to GC Unsafe is unbalanced */
-	MONO_STACKDATA (stackdata);
-	mono_threads_enter_gc_unsafe_region_unbalanced_with_info (mono_thread_info_current (), &stackdata);
+	MONO_ENTER_GC_UNSAFE;
```

Comment:
Why that change? It will break checked-build because the transition would then be unbalance.